### ISSUE TITLE
Polycyclics thoughts: on-the-fly vs group additivity

### ISSUE THREAD

**Issue by nickvandewiele** [ReactionMechanismGenerator/RMG-Java#273]
Just a post to share my experiences with thermochemistry for polycyclics.

Greg implemented the semi-empirical PM3 , molecular mechanics MM4 features. I extended Benson for polycyclics by creating a new polycyclic ring correction library. Both were fueled by the sense that something was very wrong with the thermo for polycyclic molecules.

Both have weaknesses and strengths:
- my polycyclics library has more than 100 entries, mostly based on computed or experimentally determined ring strain energies, and thus is very accurate for the enthalpy correction. S, Cp corrections for these polycyclics are far less reported, because they are not as easily determined as strain energies. Thus, i copied the S, Cp correction from the library entry that closely resembled the given polycyclic ring system. (e.g. cyclopentane is used for norbornane S, Cp correction)
- PM3: while S, Cp values for polycyclic compounds seem  to be always in the right ballpark, enthalpies sometimes do not. I have seen errors of up to 10 kcal/ mol for some molecules compared to my best estimates. The error is not systematic in the sense that many typical well-known compounds have quite accurate values. The tendency is that PM3 underestimates the enthalpy rather than overestimates it. My feeling is that PM3 sometimes goes wrong with crazy structures, created by the execution of e.g. intra_addition_exocyclic reaction family, where a cycle is created in molecules that already had some cycles.
- the PM3 implementation is very robust; RMG never hangs on a single PM3 calculation and always comes up with "something" Murat Keceli observed that PM3 sometimes gives different results for identical RMG simulations, suggesting that the PM3 calc does not always perform the same number of iterations. This resulted in significant differences.
- MM4 is an implementation, that despite its promised accuracy gain over PM3, i seldomly use. The reason is that MM4 tends to "hang" on particular molecules and completely interrupts RMG therefore. Manually killing the 1st attempt leads to the 2nd attempt (with different keywords), 3rd attempt, but these new keywords do not seem to circumvent the problem. 
- MM4 not only suffers from incidental hanging, it also "silently" continues with crazy values. E.g.:

```
Could not find a non trivial ring correction!Trying QMTP...
*****Warning: zero freqencies found (values lower than 7.7 cm^-1 are rounded to zero in MM4 output); CanTherm should hopefully correct this:
Pre-existing successful MM4 result for REXBDNBUYGXXMZ-UHFFFAOYAU (InChI=1/C8H12/c1-2-5-8-6-3-4-7-8/h2H,3-4,6-7H2,1H3) has been found. This log file will be used.
Thermo for REXBDNBUYGXXMZ-UHFFFAOYAU: 323.25    129.02  36.22   46.55   55.81   63.64   75.79   84.59   97.81   
Thermo for REXBDNBUYGXXMZ-UHFFFAOYAU_RRHO: 323.25   129.02  36.22   46.55   55.81   63.64   75.79   84.59   97.81   
Created new species: SPC(4304) InChI=1/C8H12/c1-2-5-8-6-3-4-7-8/h2H,3-4,6-7H2,1H3
ChemFormula: C8H12
1  H 0 {7,S}
2  C 0 {3,S} {9,S} {10,S} {11,S}
3  C 0 {2,S} {4,S} {12,S} {13,S}
4  C 0 {5,D} {3,S} {8,S}
5  C 0 {4,D} {6,D}
6  C 0 {5,D} {7,S} {14,S}
7  C 0 {6,S} {1,S} {15,S} {16,S}
8  C 0 {4,S} {9,S} {17,S} {18,S}
9  C 0 {2,S} {8,S} {19,S} {20,S}
10  H 0 {2,S}
11  H 0 {2,S}
12  H 0 {3,S}
13  H 0 {3,S}
14  H 0 {6,S}
15  H 0 {7,S}
16  H 0 {7,S}
17  H 0 {8,S}
18  H 0 {8,S}
19  H 0 {9,S}
20  H 0 {9,S}

The number of resonance isomers is 0
The NASA data is 
!Group:Cs-CsCsHH    Group:Cs-(Cds-Cdd-Cd)CsHH   Group:Cds-(Cdd-Cd)CsCs  Group:Cdd-CdsCds    Group:Cds-(Cdd-Cd)CsH   Group:Cs-(Cds-Cdd-Cd)HHH    Group:Cs-(Cds-Cdd-Cd)CsHH   Group:Cs-CsCsHH 
!MM4 calculation with CanTherm analysis
! [_ SMILES="InChI=1/C8H12/c1-2-5-8-6-3-4-7-8/h2H,3-4,6-7H2,1H3" _]
SPC(168)                C   8H  12          G   250.000  5000.000   995.043    1
 1.71088574E+01 3.64686093E-02-1.30374883E-05 2.10165108E-09-1.25939495E-13    2
 1.54107129E+05-4.74289947E+01 3.22989036E+00 3.56896401E-02 7.34166570E-05    3
-1.12958069E-07 4.31375788E-11 1.59669759E+05 3.35327699E+01                   4

ThermoData is 
323.25  129.02  36.22   46.55   55.81   63.64   75.79   84.59   97.81   
```

For some reason, unknown to me, the enthalpy by MM4 ends up being 323 kcal/mol while 19 kcal/ mol is a reasonable estimate.

In conclusion: 
I am thinking of creating a new thermochemistry for polycyclics strategy where enthalpies are preferrably based on polcycylic library corrections while S, and Cps are calculated on-the-fly, and thus benefitting from the strengths of both approaches. 

this should definitely improve the accuracy but will lead to longer simulation times, compared to the current strategy, where on-the-fly calcs are only done on species that did not have library entries.

Let me know what you think about this.
